prevent-broker-scale-down-if-contains-partition-replicas.md

```diff
@@ -0,0 +1,80 @@
+  We can get the desired Kafka replica count by using `kafka.getReplicas()` where `kafka` is an object of `KafkaCluster` class .
+- This method will check if the broker contains any partition replicas or not and will continue the process based on the outcome.
+- For doing so, the topic metadata will be queried to detect if the broker contains any partition replicas.
+- An Admin client instance will be used to connect with the cluster and get us the topic details(topic name and topic description)
+- Then we can use this information to check if the broker contains any partition replicas or not.
+- The scale down is done after we make sure that the brokers that are going to be removed doesn't contain any partition replicas.
+By doing this we avoid any partial scale down.
+
+### What to do if a broker contains partitions?
+
+#### Flow:
+
+- If partition replicas are found out on the broker we will revert back the Kafka replicas to the previous count by setting replicas directly in the `KafkaCluster` class using `setReplicas()` method. 
+  Changing the Kafka replica count directly in the Kafka Cluster would help us ensure that we keep same replicas everywhere like while generating certificates, services, ingresses, routes etc.
+- The broker certificates, services, ingresses, routes etc. will be treated with the original number of nodes and the rest of the reconciliation will be done normally.
+- There is a small risk at this point since the KafkaCluster instance will already exist and the storage validation will be already done. This can cause forbidden storage changes which can lead to some problems since the validation was done with the new count of Kafka replicas but at the same time the likelihood of both illegal storage changes and scale-down happening at the same time is low. 
+- We also generate a new condition which will be added to Kafka resource status depicting that the scale down is not done. It will also contain the `spec.replicas` count(which is being currently being used) in the condition message.
+  ```yaml
+  status:
+    clusterId: DoRj5f84Sruq_7TJ31y7Zw
+    conditions:
+      - lastTransitionTime: "2023-02-22T10:18:56.578009768Z"
+        message: 'Can''t Scale down since broker contains partition replicas. Ignoring
+        `replicas` setting in Kafka custom resource: my-cluster. Current Kafka replicas
+        count is 4'
+        reason: ScaleDownException
+        status: "True"
+        type: Warning
+      - lastTransitionTime: "2023-02-22T10:18:57.664668863Z"
+        status: "True"
+        type: Ready
+  ```
+
+### How to bypass the broker scale down mechanism
+
+- To bypass the broker scale down mechanism you can use the annotation `strimzi.io/bypass-brokerScaleDownCheck: "true"` on the Kafka custom resource. For e.g.
```

Comment:
It's good to have a bypass option, but I found the annotation name a bit weird. For consistency with the other annotations we provide, I would suggest `strimzi.io/bypass-broker-scaledown-check: "true"`.